### PR TITLE
Kyle backend branch

### PR DESCRIPTION
**Pull request by lamont1008**
login branch

### CODE REVIEW

File: app/exceptions/UnauthorizedException.ts
Comment (JungAn2) [inline app/exceptions/UnauthorizedException.ts:9]: Create a new exception instead of changing UnauthorizedException. Unauthorized Access is going to be used a lot for task to come.

File: app/http/controllers/AuthController.ts
Comment (JungAn2) [inline app/http/controllers/AuthController.ts:20]: Put the codes in authService not authController. The AuthController should only call function, pass in arguments and respond to the request

File: app/services/AuthService.ts
Comment (JungAn2) [inline app/services/AuthService.ts:31]: You need to return the data that is being responded back from the axios. In this line you are printing the response.data to the screen and not storing it in data(I think). Regardless you do not need to console.log the response data

File: app/services/AuthService.ts
Comment (JungAn2) [inline app/services/AuthService.ts:26]: add

http://

in front of the url

File: package.json
Comment (JungAn2) [inline package.json:41]: You probably installed this by mistake while doing npm install. Delete this

File: prisma/schema.prisma
Comment (JungAn2) [inline prisma/schema.prisma:31]: We are not storing any information of the user in local database so this schema is not needed

File: app/exceptions/UserDuplicateException.ts
Comment (JungAn2) [inline app/exceptions/UserDuplicateException.ts:1]: We are not checking for the user duplication since we are not registering anyone. Delete this file

**Review by JungAn2 — CHANGES_REQUESTED** [review]

File: app/http/controllers/AuthController.ts
Comment (wrathofrathma) [inline app/http/controllers/AuthController.ts:1]: Remove the import e.

File: app/routes/auth.ts
Comment (wrathofrathma) [inline app/routes/auth.ts:10]: Missing input validation. 

Use the zod library and the provided validation middleware. There should be examples in this project, as well as the ldap api, or go bother jung.

File: app/services/AuthService.ts
Comment (wrathofrathma) [inline app/services/AuthService.ts:3]: 👀  don't import { Axios }

File: app/services/AuthService.ts
Comment (wrathofrathma) [inline app/services/AuthService.ts:37]: Good looking out for catching the errors, but we should convert this into our own exception and throw that. I think you already defined some exceptions or have ones ready to go for this, so just look at those.

File: app/services/AuthService.ts
Comment (wrathofrathma) [inline app/services/AuthService.ts:32]: Try returning the axios value rather than storing it in data. 

In the future, make sure to initialize the values as well. 

In general, don't have console.logs you used for debugging.

File: package.json
Comment (wrathofrathma) [inline package.json:43]: Vue is for the frontend only. Delete this bad boy

File: prisma/migrations/20220330191037_init/migration.sql
Comment (wrathofrathma) [inline prisma/migrations/20220330191037_init/migration.sql:37]: remove this autogenerated shit

File: prisma/schema.prisma
Comment (wrathofrathma) [inline prisma/schema.prisma:35]: We basically don't need this entire model.

**Review by wrathofrathma — CHANGES_REQUESTED** [review]
Make sure to do unit testing for all of your functions.

File: app/schema/Auth.ts
Comment (wrathofrathma) [inline app/schema/Auth.ts:14]: This is good. 

Just keep the error strings consistent. You have punctuation in the required_error but not invalid_type_error. Super nitpicky lol. 

Also change the file name to something more descriptive, preferably something to do with the route. In the future we're going to have many files in this directory, so maybe we should establish a naming convention early. Such as RouterRouteMethod, maybe? So this would be something like AuthLoginPost.ts. This way we can quickly look at the directory and find what we're looking for or know what the schema is likely named when we're importing it.

File: .env
Comment (wrathofrathma) [inline .env]: Don't commit the .env file. I only did in the template bc i don't know how to automate the template including one, without having it just in the repo.

File: app/services/AuthService.ts
Comment (wrathofrathma) [inline app/services/AuthService.ts:29]: I'd like to see this process.env.LDAP_URL grabbed from the config folder. So we can have a default of sorts if we don't have it set as an environmental variable. Whatever default you set it to doesn't matter.

**Review by wrathofrathma — CHANGES_REQUESTED** [review]
This is good to go once you tackle the two comments
- Do environmental variable loading from the config
- Don't commit the .env file (Just delete that guy from the commit)

**Review by wrathofrathma — APPROVED** [review]

**Review by wrathofrathma — CHANGES_REQUESTED** [review]
It looks good. I'd review your unit tests one last time before merging. Some of them are failing.

**Pull request closed without merging**
